File: app/lib/data/model/thing.dart
import 'category.dart';

class Thing {
  final BigInt id;
  final String name;
  final Category category;
  final int quantity;
  final String iconUrl;
  final String imageUrl;

  const Thing({
    required this.id,
    required this.name,
    required this.category,
    required this.quantity,
    required this.iconUrl,
    required this.imageUrl,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'category': category,
      'quantity': quantity,
      'icon_url': iconUrl,
      'image_url': imageUrl,
    };
  }

  factory Thing.fromMap(Map<String, dynamic> map) {
    return Thing(
      id: BigInt.from(map['id']),
      name: map['name'] as String,
      category: Category.fromMap(Map.from(map['category'])),
      quantity: map['quantity'] as int,
      iconUrl: map['icon_url'] as String,
      imageUrl: map['image_url'] as String,
    );
  }
}

File: server/internal/domain/thing.go
package domain

type Thing struct {
	Id       int64    `json:"id,omitempty"`
	Name     string   `json:"name,omitempty"`
	Quantity int      `json:"quantity,omitempty"`
	Category Category `json:"category,omitempty"`
	ImageUrl string   `json:"image_url,omitempty"`
	IconUrl  string   `json:"icon_url,omitempty"`
}
